lightning/src/routing/scoring.rs

```diff
@@ -1270,6 +1277,12 @@ DirectedChannelLiquidity< L, HT, T> {
 		}
 
 		if score_params.probing_diversity_penalty_msat != 0 {
+			// We use `last_duration_since_epoch` as a stand-in for the current time as we don't
+			// want to fetch the current time in every score call (slowing things down
+			// substantially on some platforms where a syscall is required), don't want to add an
+			// unnecessary `std` requirement. Assuming we're probing somewhat regularly, it should
+			// reliably be close to the current time, (and using the last the last time we probed
+			// is also fine here).
```

Comment:
My thought here was to fetch time once at the start of the pathfinding operation, and just pass that around?

Reply:
If there is no method to access current time, how would `last_datapoint` be set then?

Reply:
But then the idea of passing/time-fetcher-calling at the start of pathfinding would work too?

Reply:
Isn't the additional piping worth it? It seems that in the PR as it is now you also need to add quite a bit of code to make it work, and then you still not have the actual time.

Reply:
Perhaps add it to `RouteParameters` to make the diff somewhat smaller? I am not sure if it feels right to work around not having time in the way you propose for a lightning daemon where there's a lot of time sensitivity. I'd think that making a time fetcher available where needed is better, not unlike `EntropySource`.

Reply:
Talked about this offline and decided to go with the last update time.